app/src/main/java/com/github/godspeed010/weblib/fragments/WebViewFragment.kt

```diff
@@ -93,10 +101,60 @@ class WebViewFragment : Fragment() {
                     view.findViewById<EditText>(R.id.et_address_bar).setText(url)
 
                     if (url != null) {
+                        pageError = false
+                        lastProgression = 0f
                         lastVisitedUrl = url
                     }
                 }
             }
+
+            override fun onReceivedHttpError(wv: WebView?, request: WebResourceRequest?, errorResponse: WebResourceResponse?) {
+                super.onReceivedHttpError(wv, request, errorResponse)
+
+                if (request?.url.toString() == wv?.url) pageError = true
+            }
+
+            override fun onPageFinished(wv: WebView?, loadedUrl: String?) {
+                super.onPageFinished(wv, loadedUrl)
+
+                if ((loadedUrl == novel.url) && !pageError) {
+                    if (novel.progression != 0f) {
```

Comment:
I think the purpose I had for this was so that it wouldn't unexpectedly scroll up when a user just loaded a new chapter, as the progression gets set to 0 on a new page url.